indico_toolkit/highlighter/highlighter.py

```diff
@@ -106,6 +109,127 @@ def highlight_pdf(
+            page_dimensions: {List[dict]} -- page heights and widths from ondocument OCR result, see
+                                              Ondoc class and page_heights_and_widths property
+            fill_text {dict} -- a dictionary where the keys are your labels and the val is an option from the
+                                faker library. Possible options include 'text', 'company', 'currency', 'numerify',
+                                'address', 'name', 'company_email', 'date' and many more. With 'numerify' and
+                                'text', fake data will match the length of the redacted data.
+        Example:
+            # add a key to fill_text for each label in your extraction task w/ allowed fake data method
+            fill_text = dict(member='name', birthday='date', invoice_number='numerify')
+            highlight.redact_and_replace('source.pdf', 'redacted.pdf', fill_text=fill_text)
+        """
+        num_redactions = 0
+        fake = Faker()
+        with fitz.open(self.path_to_pdf) as doc:
+            for pred in self._predictions:
+                pred_positions = [pos for pos in sorted(self._mapped_positions, key=lambda x: x["position"]["bbLeft"]) if pos["text"] == pred["text"]]
+                pred_position = pred_positions[0]
+                # Adjust bbRight to contain right bounding box of pred instead of token
+                pred_position["position"]["bbRight"] = pred_positions[-1]["position"]["bbRight"]
+                doc_page = pred_position["page_num"]
+                page = doc[doc_page]
+                xnorm = page.rect[2] / page_dimensions[doc_page]["width"]
+                ynorm = page.rect[3] / page_dimensions[doc_page]["height"]
+                annotation = self._get_annotation(pred_position["position"], xnorm, ynorm)
+                if "label" in pred and pred["label"] in fill_text:
+                    label_type = fill_text[pred["label"]]
+                    if label_type == "numerify":
+                        text = getattr(fake, label_type)(len(pred["text"]) * "#")
+                    elif label_type == "text":
+                        text = getattr(fake, label_type)(max(5, len(pred["text"])))
+                    else:
+                        text = getattr(fake, label_type)()
+                    page.add_redact_annot(
+                        annotation, text=text, fill=False, fontsize=15
+                    )
+                else:
+                    # second line of single prediction redacted
+                    page.add_redact_annot(annotation, fill=(1, 1, 1))
+                num_redactions += sum([1 for _ in page.annots()])
+                page.apply_redactions()
+            doc.save(output_path)
+        print(
+            f"*Important* to ensure that underlying data can't be recovered, convert {output_path} to a png, tif, or scanned pdf file"
+        )
+        return num_redactions
+
+    def _get_annotation(self, position: List, xnorm, ynorm):
```

Comment:
type hint xnorm, ynorm and might be worth adding a comment as to what the "inflater" is for (i.e. ensuring that we cover a bit more than the bounding box)